nix/module.nix

```diff
@@ -0,0 +1,89 @@
+inputs: { config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.asus-numberpad-driver;
+  defaultPackage = inputs.self.packages.${pkgs.system}.default;
+
+  # Function to convert configuration options to string
+  toConfigFile = cfg: builtins.concatStringsSep "\n" (
+    [ "[main]" ] ++ lib.attrsets.mapAttrsToList (key: value: "${key} = ${value}") cfg.config
+  );
+
+  # Writable directory for the config file
+  configDir = "/etc/asus-numberpad-driver";
+in {
+  options.services.asus-numberpad-driver = {
+    enable = lib.mkEnableOption "Enable the Asus Numberpad Driver service.";
+
+    layout = lib.mkOption {
+      type = lib.types.str;
+      default = "up5401ea";
+      description = "The layout identifier for the numberpad driver (e.g. up5401ea). This value is required.";
+    };
+
+    config = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      default = {};
+      description = ''
+        Configuration options for the numberpad driver.
+        These options will be written to a configuration file for the driver.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ defaultPackage ];
+
+    # Ensure the writable directory exists
+    systemd.tmpfiles.rules = [
+      "d ${configDir} 0755 root root -"
+    ];
+
+    # Write the configuration file to the writable directory
+    environment.etc."asus-numberpad-driver/numberpad_dev".text = toConfigFile cfg;
+
+    # Enable i2c
+    hardware.i2c.enable = true;
+
+    # Add groups for numpad
+    users.groups = {
+      uinput = { };
+      input = { };
+      i2c = { };
+    };
+
+    # Add root to the necessary groups
+    users.users.root.extraGroups = [ "i2c" "input" "uinput" ];
+
+    # Add the udev rule to set permissions for uinput and i2c-dev
+    services.udev.extraRules = ''
+      # Set uinput device permissions
+      KERNEL=="uinput", GROUP="uinput", MODE="0660"
+      # Set i2c-dev permissions
+      SUBSYSTEM=="i2c-dev", GROUP="i2c", MODE="0660"
+    '';
+
+    # Load specific kernel modules
+    boot.kernelModules = [ "uinput" "i2c-dev" ];
+
+    systemd.services.asus-numberpad-driver = {
+      description = "Asus Numberpad Driver";
+      after = [ "graphical-session.target" ];
+      wantedBy = [ "graphical-session.target" ];
+      serviceConfig = {
```

Comment:
@scientiac Where are stored logs on your system? I miss these lines:

```
StandardOutput=append:$ERROR_LOG_FILE_PATH
StandardError=append:$ERROR_LOG_FILE_PATH
```

In case are configured by default, the mechanism of writing to the log files is `append`?

Reply:
I have added these options now.
at:
`/var/log/asus-numberpad-driver/error.log` with `append`

Reply:
@ldrahnik For .gitignore the `result` string can be removed from there as I only used it to ensure that the built pyhton package works. I can just delete the folder before I push.

And I have removed it.